libc/src/math/generic/atan2.cpp

```diff
@@ -0,0 +1,313 @@
+//===-- Double-precision atan2 function -----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/math/atan2.h"
+#include "inv_trigf_utils.h"
+#include "src/__support/FPUtil/FPBits.h"
+#include "src/__support/FPUtil/PolyEval.h"
+#include "src/__support/FPUtil/double_double.h"
+#include "src/__support/FPUtil/multiply_add.h"
+#include "src/__support/FPUtil/nearest_integer.h"
+#include "src/__support/FPUtil/rounding_mode.h"
+#include "src/__support/macros/config.h"
+#include "src/__support/macros/optimization.h" // LIBC_UNLIKELY
+
+namespace LIBC_NAMESPACE_DECL {
+
+namespace {
```

Comment:
Actually, from the LLVM code style:

> Avoid putting declarations other than classes into anonymous namespaces

https://llvm.org/docs/CodingStandards.html#anonymous-namespaces.

Clang and LLVM Core largely conform to this, but we don't seem to follow this guideline as much in LLVM libc.

Reply:
for us, these file we only want 1 public function, the rest are static, so I think anonymous namespace provide a more clear-cut boundary, and less chance for us to forget to mark the local functions static.